src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -165,6 +170,37 @@ public ITestRun RunAsync(ITestEventListener listener, TestFilter filter)
         public void StopRun(bool force)
         {
             _engineRunner.StopRun(force);
+
+            if (force)
+            {
+                // Frameworks should handle StopRun(true) by cancelling all tests and notifying
+                // us of the completion of any tests that were running. However, this feature
+                // may be absent in some frameworks or may be broken and we may not pass on the
+                // notifications needed by some runners. In fact, such a bug is present in the
+                // NUnit framework through release 3.12 and motivated the following code.
+                //
+                // We try to make up for the potential problem here by notifying the listeners 
+                // of the completion of every pending WorkItem, that is, one that started but
+                // never sent a completion event.
+
+                if (!_workItemTracker.WaitForCompletion(WAIT_FOR_CANCEL_TO_COMPLETE))
+                {
+                    _workItemTracker.IssuePendingNotifications(_eventDispatcher);
```

Comment:
Suggestion: this method name initially confused me - I thought it was sending notifications that "the following tests are still running" 😄 How about something like `ForceSendTestCompletionEvents` or something?

Reply:
"Force" sounds odd, since the whole purpose of the class is to send notices. I like "Completion". How about "SendPendingTestCompletionEvents"?

My general problem with naming is that these reports (that's what they are actually called when received) are in two places: those received by the tracker and those sent by the tracker.